src/ui/page/share/dataset/steps/metadataFilling/metadataFilling.tsx

```diff
@@ -109,6 +128,42 @@ export const MetadataFilling: FC = () => {
     [setFormItemValue]
   )
 
+  const localizedDateFormatter = useMemo(
+    () => new Intl.DateTimeFormat(locale, { year: 'numeric', month: '2-digit', day: '2-digit' }),
+    [locale]
+  )
+
+  const handleDateRangeValueChange = useCallback(
+    (id: string) => (dateRange: DateRange) => {
+      const dateStringRange: DateStringRange = {
+        from: dateRange.from ? localizedDateFormatter.format(dateRange.from) : null,
+        to: dateRange.to ? localizedDateFormatter.format(dateRange.to) : null
+      }
+      setFormItemValue(id, dateStringRange)()
+    },
+    [setFormItemValue, localizedDateFormatter]
+  )
+
+  const dateRangeValueField = useCallback(
+    (id: string): DateRange =>
+      pipe(
+        id,
+        formItemById,
+        IOO.map(({ value }) => value as O.Option<DateStringRange>),
```

Comment:
I'm not sure it's a good idea to cast it, because when we get back other form items, we always make a predicate to check that the value we are receiving is the type we are expecting. I think we should continue this way to keep robustness

Reply:
It has been updated in 5df3d2cc. Thanks!